src/unit_api/__init__.py

```diff
@@ -0,0 +1,21 @@
+"""Unit API."""
+
+from typing import Any, Final, Protocol, Self, runtime_checkable
+
+__version__: Final = "0.0.1.dev0"
+__all__ = ["__version__", "Unit"]
+
+type Dimension = Any  # TODO: dimension-api
+type Quantity = Any  # TODO: quantity-api
+
+@runtime_checkable
+class Unit(Protocol):
+    @property
+    def dimension(self) -> Dimension: ...
+
+    def __mul__(self, other: Self, /) -> Self: ...
+    def __truediv__(self, other: Self, /) -> Self: ...
+    def __pow__(self, other: int | Self, /) -> Self: ...
```

Comment:
pint uses `NUMERIC_TYPES = (Number, Decimal, ndarray, np.number)`  https://github.com/hgrecco/pint/blob/b39b62f9cbb7c5d9589e2a3ce78deba04964a4d3/pint/compat.py#L284

astropy uses 
```
UnitPower: TypeAlias = int | float | Fraction
"""Alias for types that can be powers of the components of a
`~astropy.units.UnitBase` instance"""
UnitPowerLike: TypeAlias = UnitPower | np.integer | np.floating
"""Alias for types that can be used to create powers of the components of a
`~astropy.units.UnitBase` instance"""
```
https://github.com/astropy/astropy/blob/1477470d26611aef227a2a3dee6b6de32502f34b/astropy/units/typing.py#L71

Do we need to restrict this here?

Reply:
Implementations are free to accept other types that aren't specified. The question is: are there any uses of non-integer powers which motivate _requiring_ that they are accepted?

Reply:
When working with Qs, yes: ` xp.sqrt(Quantity(1, "km")) `.

Reply:
> Implementations are free to accept other types that aren't specified. The question is: are there any uses of non-integer powers which motivate _requiring_ that they are accepted?

Yes there's many. Getting which types to accecpt and handling this type throughout calculations is difficult, so I'd be tempted to avoid specifying it here

Reply:
extending to `int | float | self` here should be sufficient for now?

Reply:
yes, with that, am I understanding this correctly?
At mimium, a library must support int and float.  They may also support others types like Decimal.

should take out Self too

Reply:
🤦‍♂️of course, m^kg doesn't make much sense!
```suggestion
    def __pow__(self, other: int | float, /) -> Self: ...
```